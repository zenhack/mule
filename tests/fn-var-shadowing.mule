#XFAIL: type members
# This test verifies that shadowing (or lack thereof) works for function parameters.
{ type t a = (t : t a) -> int }
