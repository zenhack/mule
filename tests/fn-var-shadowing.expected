
Desugared:

	(letrec (types ((t (lam a (rec t ((t : t) -> int)))))) (values ())
	 (((.type= t) {}) (type t)))

inferred type:

	(all a (record (types ((t (lam b (rec c (c -> int)))) ...a)) (values ())))

Runtime term:

	(App ((App ((Lam (0 () (Lam (1 () (Var 1))))) (Record ()))) (Record ())))

Evaluated:

	(Record ())
